hudi-common/src/test/java/org/apache/hudi/common/table/TestTimelineUtils.java

```diff
@@ -0,0 +1,160 @@
+      String ts = i + "";
+      HoodieInstant instant = new HoodieInstant(true, HoodieTimeline.COMMIT_ACTION, ts);
+      activeTimeline.createNewInstant(instant);
+      activeTimeline.saveAsComplete(instant, Option.of(getCommitMeta(basePath, ts, ts, 2)));
+
+      HoodieInstant cleanInstant = new HoodieInstant(true, HoodieTimeline.CLEAN_ACTION, ts);
+      activeTimeline.createNewInstant(cleanInstant);
+      activeTimeline.saveAsComplete(cleanInstant, getCleanMeta(olderPartition, ts));
+    }
+
+    metaClient.reloadActiveTimeline();
+
+    // verify modified partitions included cleaned data
+    List<String> partitions = TimelineUtils.getPartitionsMutated(metaClient.getActiveTimeline().findInstantsAfter("1", 10));
+    assertEquals(5, partitions.size());
+    assertEquals(partitions, Arrays.asList(new String[]{"0", "2", "3", "4", "5"}));
+
+    partitions = TimelineUtils.getPartitionsMutated(metaClient.getActiveTimeline().findInstantsInRange("1", "4"));
+    assertEquals(4, partitions.size());
+    assertEquals(partitions, Arrays.asList(new String[]{"0", "2", "3", "4"}));
+
+    // verify only commit actions
+    partitions = TimelineUtils.getPartitionsWritten(metaClient.getActiveTimeline().findInstantsAfter("1", 10));
+    assertEquals(4, partitions.size());
+    assertEquals(partitions, Arrays.asList(new String[]{"2", "3", "4", "5"}));
+
+    partitions = TimelineUtils.getPartitionsWritten(metaClient.getActiveTimeline().findInstantsInRange("1", "4"));
+    assertEquals(3, partitions.size());
+    assertEquals(partitions, Arrays.asList(new String[]{"2", "3", "4"}));
+  }
+
+  @Test
+  public void testGetPartitionsUnpartitioned() throws IOException {
+    HoodieActiveTimeline activeTimeline = metaClient.getActiveTimeline();
+    HoodieTimeline activeCommitTimeline = activeTimeline.getCommitTimeline();
+    assertTrue(activeCommitTimeline.empty());
+
+    String partitionPath = "";
+    for (int i = 1; i <= 5; i++) {
+      String ts = i + "";
+      HoodieInstant instant = new HoodieInstant(true, HoodieTimeline.COMMIT_ACTION, ts);
+      activeTimeline.createNewInstant(instant);
+      activeTimeline.saveAsComplete(instant, Option.of(getCommitMeta(basePath, partitionPath, ts, 2)));
+
+      HoodieInstant cleanInstant = new HoodieInstant(true, HoodieTimeline.CLEAN_ACTION, ts);
+      activeTimeline.createNewInstant(cleanInstant);
+      activeTimeline.saveAsComplete(cleanInstant, getCleanMeta(partitionPath, ts));
+    }
+
+    metaClient.reloadActiveTimeline();
+
+    // verify modified partitions included cleaned data
+    List<String> partitions = TimelineUtils.getPartitionsMutated(metaClient.getActiveTimeline().findInstantsAfter("1", 10));
+    assertTrue(partitions.isEmpty());
+
+    partitions = TimelineUtils.getPartitionsMutated(metaClient.getActiveTimeline().findInstantsInRange("1", "4"));
+    assertTrue(partitions.isEmpty());
+  }
+
+  private byte[] getCommitMeta(String basePath, String partition, String commitTs, int count)
```

Comment:
Let's keep the same name as before `getCommitMetadata` & `getCleanMetadata` unless there is a specific need for keeping the names short..